drydock_provisioner/control/designs.py

```diff
@@ -25,18 +25,34 @@ class DesignsResource(StatefulResource):
 
     def __init__(self, **kwargs):
         super(DesignsResource, self).__init__(**kwargs)
-        self.authorized_roles = ['user']
 
     def on_get(self, req, resp):
+        policy_action = 'physical_provisioner:read_data'
+        ctx = req.context
         state = self.state_manager
 
-        designs = list(state.designs.keys())
-
-        resp.body = json.dumps(designs)
-        resp.status = falcon.HTTP_200
+        try:
+            if self.check_policy(policy_action, ctx):
+                designs = list(state.designs.keys())
+    
+                resp.body = json.dumps(designs)
+                resp.status = falcon.HTTP_200
+            else:
+                self.access_denied(req, resp, policy_action)
+                return
+        except Exception as ex:
```

Comment:
I do this just in case an unanticipated Exception gets thrown, I don't want a request to hang w/o a response.